vendor/bat-native-ledger/src/bat/ledger/internal/database/database_publisher_list.cc

```diff
@@ -0,0 +1,194 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/database/database_publisher_list.h"
+
+#include <tuple>
+#include <utility>
+
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/internal/database/database_util.h"
+#include "bat/ledger/internal/publisher/prefix_util.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/state/state_keys.h"
+
+using std::placeholders::_1;
+using braveledger_publisher::PrefixIterator;
+
+namespace {
+
+const char kTableName[] = "publisher_list";
+
+constexpr size_t kHashPrefixSize = 4;
+constexpr size_t kMaxInsertRecords = 100'000;
```

Comment:
In order to avoid sending the database a SQL query of unbounded length when we insert hash prefixes into the publisher list table, we divide up the work into chunks into a series of inserts with a bounded number of values.